src/osd/OSD.h

```diff
@@ -506,10 +506,14 @@ class OSDService {
   int scrubs_pending;
   int scrubs_active;
   set< pair<utime_t,spg_t> > last_scrub_pg;
+  map<spg_t, utime_t> pg_reg_stamp;
+  pair<spg_t, bool> reject_scrub_pg;
```

Comment:
This should be a boost::optional<spg_t>.  reject_scrub_pg.first is simply invalid with reject_scrub_pg.second == false.